contracts/src/spaces/entitlements/rule/RuleEntitlementV2.sol

```diff
@@ -127,7 +131,11 @@ contract RuleEntitlementV2 is
       data := add(entitlementData.offset, calldataload(entitlementData.offset))
     }
 
-    if (data.operations.length == 0) return;
+    // We should never allow the setting of empty rule datas because it can cause the xchain
```

Comment:
This is the part that would have caused a revert before, but I'm adding reverts everywhere we don't set an entitlement to create an invariant that setEntitlement always sets an entitlement. This is already true for UserEntitlement.